app/src/androidTest/java/com/github/fribourgsdp/radio/unit/HostGameHandlerTest.kt

```diff
@@ -151,4 +154,60 @@ class HostGameHandlerTest {
         assertTrue(view.gameOver)
         assertTrue(view.crashed)
     }
+
+    @Test
+    fun incrementationOfRoundTest(){
+        val view = FakeGameView()
+        val game = Game.Builder()
+            .setNoSing(true)
+            .setHost(host)
+            .addUserId(otherPlayer.id)
+            .setPlaylist(Playlist("Test PL", setOf(Song("a", "b", "c")), Genre.NONE))
+            .build()
+        val db = mock(Database::class.java)
+        `when`(db.updateGame(anyLong(), anyMap()))
+            .thenReturn(Tasks.forResult(null))
+        `when`(db.resetGameMetadata(anyLong(), anyString()))
+            .thenReturn(Tasks.forResult(null))
+        `when`(db.updateCurrentSongOfGame(anyLong(), anyString(), anyLong()))
+            .thenReturn(Tasks.forResult(null))
+        val handler = HostGameHandler(ctx, game, view, db, noSing = true)
+        handler.handleSnapshot(mockSnapshot)
+
+        // Wait for the task of the database to execute
+        Thread.sleep(sleepingTime)
```

Comment:
this is not very good... why do we need this since we are mocking the database?

Reply:
If followed the scheme of the other tests of HosGameHandlerTest, but I can try to see if it works without it.
The fact is that even though it is mocked, handleSnapshot still has asynchronous side-effects.

Reply:
ah ok in this case I understand.